### PR TITLE
feat(kit): allow configuration of default options for input-count

### PR DESCRIPTION
**Pull request by KarimovDev**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Partially implements #133

## What is the new behavior?

Default input-count options can be set via a provider. This is a partial implementation of #133

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/kit/components/input-count/input-count-options.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {InjectionToken} from '@angular/core';
+import {TuiContextWithImplicit} from '@taiga-ui/cdk';
+import {TuiSizeL} from '@taiga-ui/core';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+
+export interface InputCountOptions {
+    readonly icons: Readonly<{
+        up: PolymorpheusContent<TuiContextWithImplicit<TuiSizeL>>;
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count-options.ts:8]: Buttons are rather tiny, they always have size "s". How about we simplify it by keeping context `{}` — this way you can just pass it to buttons' `icon` input, as it is already polymorpheus.

File: projects/kit/components/input-count/input-count-options.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export const PASSWORD_ICON_DOWN = 'tuiIconMinus';
+
+/** Default values for the input password options. */
+export const TUI_INPUT_COUNT_DEFAULT_OPTIONS: InputCountOptions = {
+    icons: {
+        up: PASSWORD_ICON_UP,
+        down: PASSWORD_ICON_DOWN,
+    },
+    appearance: 'textfield',
+    hideButtons: false,
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count-options.ts:30]: Right now we force hide buttons in tables by comparing appearance to table. That's rather hardcode. @vladimirpotekhin  @nsbarsukov what do you think, it probably makes sense to alter this token in `Table` directive now? With a factory, skipSelf and spread with `hideButtons` set to `true`? Sounds like that would be more idiomatic.

File: projects/kit/components/input-count/input-count-options.ts (hunk 1 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts:8]: But buttons' height gets smaller when input-count uses "s" size. And this polymorpheus context is taken into account textfield size

File: projects/kit/components/input-count/input-count-options.ts (hunk 2 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts:30]: In this case someone can rewrite token if they use nested components inside the table. Or am I wrong?

Comment (waterplea) [reply projects/kit/components/input-count/input-count-options.ts:30]: In this case, if somebody uses component inside table and for some reason want to change it, they can create a directive that would override it, yes. I don't think that's bad.

File: projects/kit/components/input-count/input-count-options.ts (hunk 1 above)
Comment (waterplea) [reply projects/kit/components/input-count/input-count-options.ts:8]: I meant that you probably cannot fit a big icon anyway, even for a big input size. Template complexity really increases now and I don't want this token option feature to become an overkill.

File: projects/kit/components/input-count/input-count-options.ts (hunk 2 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts:28]: Ok. I've already done this as an example. @vladimirpotekhin @nsbarsukov you can check it right here

File: projects/kit/components/input-count/input-count-options.ts (hunk 1 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts]: Ok. I agree about template complexity. I've changed it to {} context

File: projects/kit/components/input-count/input-count.component.ts [hunk 3]
```diff
@@ -118,7 +126,7 @@ export class TuiInputCountComponent
 
     @HostBinding('class._has-buttons')
     get hasButtons(): boolean {
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count.component.ts:128]: Unnecessary getter, let's revert what CSS does and bind it to input directly

Comment (KarimovDev) [reply projects/kit/components/input-count/input-count.component.ts]: Done

File: projects/kit/components/input-count/input-count-options.ts (hunk 1 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts]: @waterplea there is also tuiTableMode directive with TUI_TEXTFIELD_APPEARANCE token. Should we add TUI_INPUT_COUNT_OPTIONS here as well? In this case should we move TUI_INPUT_COUNT_OPTIONS into core package?

File: projects/kit/components/input-count/input-count-options.ts (hunk 2 above)
Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts:28]: @waterplea there is also tuiTableMode directive with TUI_TEXTFIELD_APPEARANCE token. Should we add TUI_INPUT_COUNT_OPTIONS here as well? In this case, should we move TUI_INPUT_COUNT_OPTIONS into core package?

Comment (waterplea) [reply projects/kit/components/input-count/input-count-options.ts:28]: No, that directive was there before we introduced table component. I believe we might end up just dropping it in the next major release.

**Review by waterplea — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-count/input-count-options.ts (hunk 2 above)
Comment (waterplea) [reply projects/kit/components/input-count/input-count-options.ts:28]: Oops, @KarimovDev I see this directive is still actually in use on the demo and that fails the screenshots. Let's copy-paste this factory for the time being then.

Comment (KarimovDev) [reply projects/kit/components/input-count/input-count-options.ts:28]: Done

**Review by waterplea — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**